engine/api/router_middleware_auth_permission.go

```diff
@@ -321,7 +333,12 @@ func (api *API) checkGroupPermissions(ctx context.Context, w http.ResponseWriter
 		return sdk.WithStack(sdk.ErrForbidden)
 	}
 
-	if permissionValue > sdk.PermissionRead { // Only group administror or CDS administrator can update a group or its dependencies
+	if permissionValue > sdk.PermissionRead {
+		// Hatcheries started for "shared.infra" group are granted for group "shared.infra"
+		if g.ID == group.SharedInfraGroup.ID && isHatchery(ctx) && isGroupMember(ctx, g) {
```

Comment:
I think we should have a helper like isHatcheryShared that return true if the consumer contains the shared.infra group (having shared.infra in a consumer should let the hatchery book/register any models, maybe we should create a dedicated middleware for the route like /book on worker model with that behavior and use the existing one only for user actions).
Here isGroupMember will return true for every user as we want everyone to be able to RO shared.infra resources. We can't use this one to check write permission on a group as this will let everyone edit shared.infra resources.